aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -277,18 +277,24 @@ Tensor empty_like(
     "Cannot set memory_format both in TensorOptions and explicit argument; please delete "
     "the redundant setter.");
 
-  TensorOptions options =
-      self.options()
-          .merge_in(options_)
-          .merge_in(TensorOptions().memory_format(optional_memory_format));
-
-  if (options.layout() == kSparse && self.is_sparse()) {
-    auto result = at::empty({0}, options); // to be resized
+  // Avoid adding optional_memory_format to options_ because it can be invalid for non-strided tensors.
+  if (options_.layout() == kSparse && self.is_sparse()) {
+    if (optional_memory_format.has_value()) {
+      TORCH_CHECK(
+        optional_memory_format == MemoryFormat::Preserve,
+        "This memory format option is not supported by sparse tensors");
```

Comment:
I wonder if we could improve the exception message to be more informative.
For instance, use
```
TORCH_CHECK(optional_memory_format == MemoryFormat::Preserve,
            "Memory format for sparse tensors must be Preserve, got ", optional_memory_format);